test/DynamoCoreWpfTests/RecordedTests.cs

```diff
@@ -3827,6 +3829,1109 @@ public void Lacing_Deffect_5759()
+               else if (commandTag == "SecondRun")
+                {
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("67ef5890-7aa6-4db6-ae82-c4532efc0c01");// check cbn 
+                    Assert.AreEqual(ElementState.Active, node.State);
+                    AssertPreviewCount("67ef5890-7aa6-4db6-ae82-c4532efc0c01", 3);
+                   
+                }
+                else if (commandTag == "Tag-ca842c63")
+                {
+                  
+                     Assert.IsNotNull(GetPreviewValue("3c1cfb55-f23b-4a5b-9dc7-f02fde9215a4"));//check write to excel node
+                     AssertPreviewCount("3c1cfb55-f23b-4a5b-9dc7-f02fde9215a4", 65);
+                }
+            });
+        }
+
+
+        [Test]
+        public void MAGN_7348_Core_Python()
+        {
+            // Check Python nodes
+            // Scenario
+            //  a) By connecting node
+            //  b) change python script
+            //  c) reconnect/invalid input
+            // http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7348
+
+            preloadGeometry = true;
+            RunCommandsFromFile("MAGN_7348_Core_Python.xml", (commandTag) =>
+            {
+                var workspace = ViewModel.Model.CurrentWorkspace;
+
+                if (commandTag == "FirstRun")
+                {
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("8a747cdc-7891-4c8a-bfb6-63ad5fbc54c3");// check list.GetItemAtIndex 
+                    Assert.AreEqual(ElementState.Warning, node.State);
+                }
+                else if (commandTag == "SecondRun")
+                {
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("8a747cdc-7891-4c8a-bfb6-63ad5fbc54c3");// check list.GetItemAtIndex 
+                    Assert.AreEqual(ElementState.Active, node.State);
+                    AssertPreviewCount("8a747cdc-7891-4c8a-bfb6-63ad5fbc54c3", 11);
+                    for (int i = 0; i < 11; i++)
+                    {
+                        Assert.IsNotNull(GetPreviewValueAtIndex("8a747cdc-7891-4c8a-bfb6-63ad5fbc54c3", i));
+                    }
+                }
+                else if (commandTag == "ThirdRun")
+                {
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("8a747cdc-7891-4c8a-bfb6-63ad5fbc54c3");// check list.GetItemAtIndex 
+                    Assert.AreEqual(ElementState.Warning, node.State);
+                     NodeModel python = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("3617330e-a33d-4fd6-b13d-b3eee9da5f18");// check python script
+                    Assert.AreEqual(ElementState.Warning, python.State);
+                    
+                }
+
+            });
+        }
+
+
+        [Test,Category("Failure")]
```

Comment:
this is the additional input portals problem.
extra inputs are generated, which is wrong